datacapture/src/main/java/com/google/android/fhir/datacapture/extensions/MoreQuestionnaires.kt

```diff
@@ -73,38 +74,46 @@ internal fun validateLaunchContextExtensions(launchContextExtensions: List<Exten
   }
 
 /**
- * Checks that the extension:name extension exists and its value contains a valid code from
- * [QuestionnaireLaunchContextSet]
+ * Verifies the existence of extension:name and extension:type with valid system and type values
+ * from the [QuestionnaireLaunchContextSet].
  */
 private fun validateLaunchContextExtension(launchExtension: Extension) {
   check(launchExtension.extension.size == 2) {
     "The extension:name or extension:type extension is missing in $EXTENSION_SDC_QUESTIONNAIRE_LAUNCH_CONTEXT"
   }
 
-  val isValidExtension =
-    QuestionnaireLaunchContextSet.values().any {
-      launchExtension.equalsDeep(
+  val nameCoding = launchExtension.getExtensionByUrl("name").value as Coding
```

Comment:
does the block of code below correctly validate the launch context extension? if so, maybe we should just do it like so 
```suggestion
  val nameCoding = launchExtension.getExtensionByUrl("name").value as Coding
  val typeCodeType = launchExtension.getExtensionByUrl("type").value as CodeType

  val isValidExtension = QuestionnaireLaunchContextSet.values().any { context ->
    nameCoding.code == context.code &&
    nameCoding.display == context.display &&
    typeCodeType.value == context.resourceType &&
    launchExtension.getExtensionByUrl("name").value.system == context.system
  }
```

Reply:
I propose something like this
1. since we are deprecating QuestionnaireLaunchContextSet
2. only validates the nameCoding system and type resourcetype
```suggestion
  val nameCoding = launchExtension.getExtensionByUrl("name").value as Coding
  val typeCodeType = launchExtension.getExtensionByUrl("type").value as CodeType

  val hasNameCodingSystem = nameCoding.system == EXTENSION_LAUNCH_CONTEXT
  val isValidResourceType = MoreResourceType.isValidResourceType(typeCodeType.value)

  val isValidExtension = hasNameCodingSystem && isValidResourceType
```